go/vt/vtgate/buffer/buffer.go

```diff
@@ -94,6 +94,18 @@ func CausedByFailover(err error) bool {
 	return isFailover
 }
 
+// IsErrorDueToReparenting is a stronger check than CausedByFailover, meant to return
+// if the failure is caused because of a reparent.
+func IsErrorDueToReparenting(err error) bool {
```

Comment:
This seems to be used only within the `buffer` package. It does not need to be exported.